### PR TITLE
Remote Transport - Firestore

### PR DESCRIPTION
**Pull request by Neche-Stephen**
Initialized firebase and set up transports for firestore

### CODE REVIEW

File: src/transports.ts [hunk 1]
```diff
@@ -13,6 +18,21 @@ export const fileTransport = async (logEntry: LogEntry) => {
   await fs.appendFile(logFilePath, JSON.stringify(logEntry) + "\n");
 };
 
-export const remoteLogsTransport = (logEntry: LogEntry) => {
-  // upload to remote logs storage service like sentry
+
+
+export const remoteLogsTransport = async (logEntry: LogEntry) => {
+  if (!logEntry.requestId) {
```
Comment (justiceo) [inline src/transports.ts]: Move these to a log function inside firebase.ts

Rename the firebase directory to transports and put other transports there, each in its own file with a log function.

File: src/firebase/firebase.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+// Import the functions you need from the SDKs you need
+import { initializeApp } from "firebase/app";
+import { getFirestore, doc, setDoc, Firestore } from "firebase/firestore"; 
+import dotenv from 'dotenv';
+dotenv.config();
+
+const firebaseConfig = {
+  apiKey: process.env.FIREBASE_API_KEY,
```
Comment (justiceo) [inline src/firebase/firebase.ts]: Check in an env file or yaml file that contains all the keys you've used so far, with empty values.

That way, there is no need to browse the entire repository to determine which keys need to be provided and from where.

**Review by justiceo — CHANGES_REQUESTED** [review]

File: src/transports/firebase.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    locale: "en-US",
+    userTimeZone: "America/Los_Angeles",
+    operatingSystem: "darwin/23.1.0",
+    shell: "/bin/zsh",
+    memory: 0,
+    machineId:
+      "b54b5520aaabea2a154763eeff5ba3cf72c05c576d81d9bd37aa8714e717de20",
+    env: "test",
+  },
+});
```
Comment (justiceo) [inline src/transports/firebase.ts]: move this to a spec file

File: src/transports/transports.ts [hunk 4]
```diff
@@ -1,16 +1,17 @@
...
-import { db } from "./firebase/firebase";
-import { doc, setDoc } from "firebase/firestore";
-
-import dotenv from "dotenv";
-dotenv.config();
 
 // Local dependencies
 import { LogEntry } from "../log-entry";
 
+// Third party packages
```
Comment (justiceo) [inline src/transports/transports.ts:8]: These imports don't seem used. I also don't see the utility of this file, okay to delete.

File: config.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+
+
+OPENAI_API_KEY=""
```
Comment (justiceo) [inline config.yaml:3]: Can you add the usage of this config alongside it in the same PR?

**Review by justiceo — CHANGES_REQUESTED** [review]

File: src/transports/transports.ts (hunk 4 above)
Comment (Neche-Stephen) [reply src/transports/transports.ts:8]: They are used. The file is where we are going to implement transport for json array, and where consoleTransport and fileTransport

File: src/transports/firebase.ts (hunk 3 above)
Comment (Neche-Stephen) [reply src/transports/firebase.ts]: This has been done. It is an outdated commit

File: config.yaml (hunk 5 above)
Comment (Neche-Stephen) [reply config.yaml:3]: Alright. will do.

File: src/transports/firebase.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  projectId: process.env.FIREBASE_PROJECT_ID,
+  storageBucket: process.env.FIREBASE_STORAGE_BUCKET,
+  messagingSenderId: process.env.FIREBASE_MESSAGING_SENDER_ID,
+  appId: process.env.FIREBASE_APP_ID,
+  measurementId: process.env.FIREBASE_MEASUREMENT_ID,
+};
+
+// Initialize Firebase
+const app = initializeApp(firebaseConfig);
+export const db: Firestore = getFirestore(app);
```
Comment (justiceo) [inline src/transports/firebase.ts:24]: Move these to the `send` function. I don't expect 'send' to be invoked often but these objects would be initialized with every import as-is.

**Review by justiceo — CHANGES_REQUESTED** [review]

**Pull request closed without merging**